src/components/appSwitch/AppSwitch.js

```diff
@@ -241,24 +238,31 @@ const AppSwitch = (props: LoadingProps) => {
 
   useEffect(() => {
     if (isMobileNative && DeepLinking.pathname) {
-      return props.navigation.navigate(DeepLinking.pathname.slice(1))
+      deepLinkingNavigation()
     }
   }, [])
 
   useEffect(() => {
     if (isMobileNative && DeepLinking.pathname) {
       DeepLinking.subscribe(() => {
-        return props.navigation.navigate(DeepLinking.pathname.slice(1))
+        deepLinkingNavigation()
       })
     }
+    return () => {
+      DeepLinking.unsubscribe()
+    }
   }, [DeepLinking.pathname])
 
   useEffect(() => {
-    AppState.addEventListener('change', handleAppFocus)
-    return function() {
-      AppState.removeEventListener('change', handleAppFocus)
+    if (ready && gdstore && appState === 'active') {
+      checkBonusInterval(true)
+      showOutOfGasError(props)
+      if (isMobileNative && !initSubscribe.current) {
+        initSubscribe.current = true
+        DeepLinking.subscribe(deepLinkingNavigation())
```

Comment:
also this is incorrect as deepLinkingNavigation is the subscription callback, not a callback factory
btw it should be removed (see my comment above)